src/string.cr

```diff
@@ -539,16 +543,31 @@ class String
       ptr += 1
 
       if prefix
-        case ptr.value.unsafe_chr
-        when 'b'
-          base = 2
-          ptr += 1
-        when 'x'
-          base = 16
-          ptr += 1
-        else
-          base = 8
-        end
+        found_digit = case ptr.value.unsafe_chr
```

Comment:
I'd refactor this to create a new variable with default (`found_digit = false`), then assign it to `true` in just one branch, where it's needed.

Reply:
Yup, that code is silly. I was too fixated on using a return value from the case statement.